### ISSUE TITLE
How to configure for GHE?

### ISSUE THREAD

**Issue by ScottChapman** [OneGraph/graphiql-explorer#54]
I love this, and wanted to set this up using our own github enterprise deployment, but I couldn't figure out how to configure the endpoint. Is that documented somewhere?

**Comment by FluorescentHallucinogen** [comment]
@ScottChapman See https://github.com/OneGraph/graphiql-explorer-example.

I agree, it would be nice to have a detailed description of each prop (something like https://github.com/OneGraph/graphiql-code-exporter#props) or at least a link to https://github.com/OneGraph/graphiql-explorer-example in the ReadMe.

**Comment by FluorescentHallucinogen** [comment]
Oh, the ReadMe already have the link to https://github.com/OneGraph/graphiql-explorer-example, see https://github.com/OneGraph/graphiql-explorer#example-usage.

**Comment by ScottChapman** [comment]
Thanks, I actually did read through those. They seemed like mostly UI properties, not really related to setting a different graphql endpoint with auth headers.

**Comment by ScottChapman** [comment]
I think maybe I figured it out, I poked around in `src/App.js` and found some entries for the fetcher. I modified those by changing the URL and adding an authorization bearer token, and that worked for me.
Was that the proper way for me to customize it?

**Comment by FluorescentHallucinogen** [comment]
@ScottChapman Yes, https://github.com/OneGraph/graphiql-explorer-example/blob/master/src/App.js#L17 is the only line you should change. You can also add headers here: https://github.com/OneGraph/graphiql-explorer-example/blob/master/src/App.js#L20. Feel free to extract them to separate variables and use object spread to merge headers. :wink:

**Comment by ScottChapman** [comment]
coolio!

So, if I wanted to deploy locally here, and make it so that other users would need to authenticate with our GHE (meaning I wouldn't hard code my bearer token), is there an easy way to do that? Or I suspect that might be beyond the scope of the example?

**Comment by FluorescentHallucinogen** [comment]
`<GraphiQLExplorer>` component uses interface very similar to interface of `<GraphiQL>` component, both have the `schema` prop. You should provide the same value to both `schema` so that everything works correctly. Both `<GraphiQLExplorer>` and `<GraphiQL>` components don't care how and where you get the `schema` prop's value.

> Or I suspect that might be beyond the scope of the example?

Yes, providing UI for headers is out of scope of these projects.

PTAL at https://graphiql-online.com. It's GraphiQL + Custom Headers plugin + GraphiQL Explorer + GraphQL Voyager. :wink:

**Comment by ScottChapman** [comment]
oh nice! (except Voyager comes up blank for me)

**Comment by FluorescentHallucinogen** [comment]
> except Voyager comes up blank for me

Hm, that's strange! But GraphiQL and Explorer works? Could you please share the URL of your endpoint? I'll see what the problem is.

**Comment by ScottChapman** [comment]
https://github.ibm.com/api/graphql

I can send you console logs

**Comment by FluorescentHallucinogen** [comment]
Could you please provide auth header value for tests?

**Comment by ScottChapman** [comment]
nope, it's our company's enterprise github deployment...

**Comment by FluorescentHallucinogen** [comment]
You can revoke this token later. Otherwise, ¯\_(ツ)_/¯.

**Comment by ScottChapman** [comment]
yea, sorry... What's the above error look like? Is it running some query against the end point? Perhaps I can run it and send you the results?

**Comment by FluorescentHallucinogen** [comment]
Try https://graphql-voyager.web.app ("Change schema" button -> "URL" tab). If it works in standalone GraphQL Voyager, this is a bug in GraphiQL Online.

**Comment by ScottChapman** [comment]
yes, that worked...
